### DISCUSSION TITLE
SQSTemplate error handling

### DISCUSSION THREAD

**Discussion by lazar725** [awspring/spring-cloud-aws#1094] [Q&A]
Question:

I'm using spring-cloud-aws v3.1.0 in a Spring Boot app v3.2.1.
What is the proper way of handling errors when sending a message to SQS fails? Is there a specific exception I should catch?

    sqsTemplate
            .send(sqsSendOptions ->
                            sqsSendOptions
                                    .queue(queueName)
                                    .payload(message)
                    )

**Comment by tomazfernandes** [comment]
@lazar725 , you can catch a `MessagingOperationFailedException`. Inside it you'll find the original AWS SDK exception.

How to specifically handle the error depends on your actual requirements.

Does this answer your question?

**Reply by lazar725** [reply]
@tomazfernandes Thanks for your help, it does answer the question. I'm just wondering why in all the SQSTemplate tutorials they simply ignore error handling when sending messages to SQS. :)

**Reply by tomazfernandes** [reply]
Well, I guess SqsTemplate doesn't have anything that requires a special error treatment

When you send a message, that might throw an exception, and your logic should be prepared to handle that, just like writing to a DB might fail, or an HTTP request.

Do you have anything different in mind?

**Reply by lazar725** [reply]
A logic should definitely be prepared to handle that, the missing part was which exception to handle.

**Reply by tomazfernandes** [reply]
Oh, I see, makes sense.

Would you like to open a PR to add that information to the docs?
